packages/@aws-cdk/aws-codedeploy/lib/ecs/deployment-group.ts

```diff
@@ -32,14 +46,144 @@ export interface IEcsDeploymentGroup extends cdk.IResource {
 }
 
 /**
- * Note: This class currently stands as a namespaced container for importing an ECS
- * Deployment Group defined outside the CDK app until CloudFormation supports provisioning
- * ECS Deployment Groups. Until then it is closed (private constructor) and does not
- * extend {@link Construct}.
+ * Details about an ECS service and its corresponding ECS cluster
+ */
+export type EcsService = ecs.Ec2Service | ecs.FargateService | ecs.ExternalService;
+
+/**
+ * A route with an ELB listener and target group.
+ */
+export interface EcsTrafficRoute {
```

Comment:
I find this coupling of the listener and the target group confusing, because of how CodeDeploy orchestrates blue vs green task sets and the traffic shifting for ECS.  CodeDeploy requires 2 target groups (one blue, one green), but only requires a prod listener.  CodeDeploy will shift the prod listener from blue to green, so the listener is not tied to a specific target group.

Reply:
That make sense. I got rid of `TrafficRoute` interface and kept as 4 separate attributes (2 TGs and 2 listeners)